tests/core/test_expectation_suite.py

```diff
@@ -26,7 +33,290 @@ def exp1():
+            "batch_spec",
+            "batch_markers",
+        }
+        # batch_definition (along with other keys that are not provided) should be filtered out
+        assert "batch_definition" not in citation_keys
+
+    @pytest.mark.unit
+    @pytest.mark.v2_api
+    def test_add_citation_accepts_v2_api_params(
+        self, empty_suite_with_meta: ExpectationSuite
+    ):
+        """This test ensures backward compatibility with the v2 api and can be removed when deprecated."""
+        empty_suite_with_meta.add_citation(
+            "fake_comment",
+            batch_kwargs={"fake": "batch_kwargs"},
+            batch_parameters={"fake": "batch_parameters"},
+        )
+
+        assert empty_suite_with_meta.meta["citations"][0]["batch_kwargs"] == {
+            "fake": "batch_kwargs"
+        }
+        assert empty_suite_with_meta.meta["citations"][0]["batch_parameters"] == {
+            "fake": "batch_parameters"
+        }
+        citation_keys = set(empty_suite_with_meta.meta["citations"][0].keys())
+        assert citation_keys == {
+            "comment",
+            "citation_date",
+            "batch_kwargs",
+            "batch_parameters",
+        }
+
+    @pytest.mark.unit
+    def test_add_citation_all_params(self, empty_suite_with_meta: ExpectationSuite):
+        """Note, this does not include v2 api params e.g. batch_kwargs"""
+        empty_suite_with_meta.add_citation(
+            "fake_comment",
+            batch_request={"fake": "batch_request"},
+            batch_definition={"fake": "batch_definition"},
+            batch_spec={"fake": "batch_spec"},
+            batch_markers={"fake": "batch_markers"},
+            profiler_config={"fake": "profiler_config"},
+            citation_date="2022-09-08",
+        )
+        assert "citations" in empty_suite_with_meta.meta
+
+        citation_keys = set(empty_suite_with_meta.meta["citations"][0].keys())
+        assert citation_keys == {
+            "comment",
+            "citation_date",
+            "batch_request",
+            "batch_definition",
+            "batch_spec",
+            "batch_markers",
+            "profiler_config",
+        }
+
+    @pytest.mark.unit
+    def test_add_citation_citation_date_str_override(
+        self, empty_suite_with_meta: ExpectationSuite
+    ):
+        empty_suite_with_meta.add_citation("fake_comment", citation_date="2022-09-08")
+        assert (
+            empty_suite_with_meta.meta["citations"][0]["citation_date"]
+            == "2022-09-08T00:00:00.000000Z"
+        )
+
+    @pytest.mark.unit
+    def test_add_citation_citation_date_datetime_override(
+        self, empty_suite_with_meta: ExpectationSuite
+    ):
+        empty_suite_with_meta.add_citation(
+            "fake_comment", citation_date=datetime.datetime(2022, 9, 8)
+        )
+        assert (
+            empty_suite_with_meta.meta["citations"][0]["citation_date"]
+            == "2022-09-08T00:00:00.000000Z"
+        )
```

Comment:
Might it make sense to parameterize to consolidate the logic of these two?

Reply:
Good call! Done.